rlclientlib/extensions/onnx/src/onnx_input.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include <core/session/onnxruntime_cxx_api.h>
+
+#include "api_status.h"
+
+namespace reinforcement_learning { namespace onnx {
+  using byte_t = unsigned char;
+  using bytes_t = std::vector<unsigned char>;
+  using tensor_data_t = std::pair<bytes_t, bytes_t>;
+
+  /**
+   * Check whether the provided bytes contain enough data to represent an
+   * array of elements of type element_t
+   */
+  template <typename element_t>
+  inline bool check_array_size(const bytes_t& bytes, size_t& element_count)
+  {
+    element_count = bytes.size() / sizeof(element_t);
```

Comment:
shouldn't we have bytes.size() * sizeof(byte_t) instead of bytes.size() here? (probably not critical right now)

Reply:
Not critical, but might as well make it safe.